products/jbrowse-cli/src/base.ts

```diff
@@ -52,6 +121,14 @@ export default abstract class JBrowseCommand extends Command {
     return fsPromises.readFile(location, { encoding: 'utf8' })
   }
 
+  async writeJsonConfig(config: string) {
+    try {
+      fsPromises.writeFile('./config.json', config)
+    } catch (error) {
+      this.error(`${error}`)
```

Comment:
You can actually pass the error straight to the error logger without stringifying it: https://oclif.io/docs/commands#thiserrormessage-string--error-options-code-string-exit-number-ref-string-suggestions-string
```suggestion
      this.error(error)
```

Reply:
This is a good point but it seems we should really not have to have these manual calls to this.error though? The default behavior of the oclif would be to catch this error and report it. I think more of our CLI code should rely on this instead of doing our manual catch-and-report

Reply:
Okay, I modeled the `this.error('Could not write config file')` after other parts of the CLI code, like `this.error(
        'Could not find the file "manifest.json". Please make sure you are in the top level of a JBrowse 2 installation.',
        { exit: 10 },
      )` in the checkLocation() function of the base class. This way, you are passing an explicit string that gives the user an indication of what caused the error. Would the default oclif error messages be this readable?

Reply:
That makes sense to base it off the existing code, but the existing code may need work. Personally, I think it is better to use the default error handling of oclif with the caught exception. This way we will get a stack trace this way, and it aids accurate error reporting more than providing a custom error message does

I made a short proposal for this here #1239